src/evm/Readme.md

```diff
@@ -2,73 +2,72 @@
 
 Simulate any on-chain transaction, overriding the world state if needed.
 
-`simulation::SimulationEngine` can simulate an on-chain transaction and return tx output and state changes caused by
-that transaction.
+The `SimulationEngine` leverages `revm` to simulate on-chain transactions, returning both transaction output and state
+changes caused by the transaction. I also supports overriding account balances and storage slots.
 
-To do that, the engine needs access to the EVM state. This is provided by `database::SimulationDB`. This struct queries
-an ethereum node for needed data and caches it for a duration of 1 block. It also allows for setting up mocked accounts
-and overriding account balances and storage slots.
+To perform simulations, the engine requires access to the EVM state, provided by the `EngineDatabaseInterface` trait.
+Two
+implementations of this trait are available:
```

Comment:
Is it just looking weird because of the github UI? Probably
```suggestion
Two implementations of this trait are available:
```

Reply:
oh this is weird..